grype/version/kb_contraint.go

```diff
@@ -37,14 +37,10 @@ func (c kbConstraint) supported(format Format) bool {
 }
 
 func (c kbConstraint) Satisfied(version *Version) (bool, error) {
-	if c.raw == "" && version != nil {
```

Comment:
nit: since this is surprising relative to other constraints defined, can we add more details in comments as to why this is different?